server/server.js

```diff
@@ -1,90 +1,169 @@
-const express = require('express');
-const cors = require('cors');
-require('dotenv').config();
-const path = require('path');
-const db = require('./db/db-connection.js');
-
+const express = require("express");
+const cors = require("cors");
+require("dotenv").config();
+const path = require("path");
+const db = require("./db/db-connection.js");
+const { auth } = require("express-oauth2-jwt-bearer");
 
 const app = express();
+
+const REACT_BUILD_DIR = path.join(__dirname, "..", "client", "dist");
+app.use(express.static(REACT_BUILD_DIR));
+
 const PORT = process.env.PORT || 8080;
 app.use(cors());
 app.use(express.json());
 
-// creates an endpoint for the route "/""
-app.get('/', (req, res) => {
-    res.json({ message: 'Hola, from My template ExpressJS with React-Vite' });
+app.get("/", (req, res) => {
+  try {
+    res.sendFile(path.join(REACT_BUILD_DIR, "index.html"));
+  } catch (error) {
+    console.log(error.message);
+  }
 });
 
-// create the get request for students in the endpoint '/api/students'
-app.get('/api/students', async (req, res) => {
-    try {
-        const { rows: students } = await db.query('SELECT * FROM students');
-        res.send(students);
-    } catch (e) {
-        return res.status(400).json({ e });
-    }
+// add new user to DB
+app.post("/api/addUser", async (req, res) => {
+  try {
+    const { userId, userEmail } = req.body;
+    const { rows: user } = await db.query(
+      "INSERT INTO users(id, email) VALUES($1, $2) ON CONFLICT DO NOTHING RETURNING*",
+      [userId, userEmail]
+    );
+    user[0] && console.log("User added:", user[0]);
+    res.status(200).json(user);
+  } catch (error) {
+    console.log(error.message);
+  }
 });
 
-// create the POST request
-app.post('/api/students', async (req, res) => {
-    try {
-        const newStudent = {
-            firstname: req.body.firstname,
-            lastname: req.body.lastname,
-            iscurrent: req.body.iscurrent
-        };
-        //console.log([newStudent.firstname, newStudent.lastname, newStudent.iscurrent]);
-        const result = await db.query(
-            'INSERT INTO students(firstname, lastname, is_current) VALUES($1, $2, $3) RETURNING *',
-            [newStudent.firstname, newStudent.lastname, newStudent.iscurrent],
-        );
-        console.log(result.rows[0]);
-        res.json(result.rows[0]);
+// add video
+app.post("/api/addVideo/:videoId", async (req, res) => {
```

Comment:
Hmm interested in hearing why you chose a separate addVideo endpoint (rather than including the add video logic inside the addWorkout endpoint) ?